### ISSUE TITLE
Properly ignore deleted enums, messages, and services in unstable packages if their corresponding file is deleted

### ISSUE THREAD

**Issue by chronitis** [bufbuild/buf#412]
This is related to #349 (and transitively #37 and #325); `buf` version 0.49.0.

With `ignore_unstable_packages`, deletion of files in an unstable package is allowed (as I would expect) with `breaking.use: [FILE]`. However, it is still an error with `breaking.use: [PACKAGE]` to delete part of a package, but not an entire package.

I would expect the same behaviour in either case (that deletion from an unstable package is not a breaking change for either `FILE` or `PACKAGE` categories, regardless of whether a package is partially or wholly deleted.

With an (unstable) package containing two messages committed to git:

```
test/v1test/a.proto
test/v1test/b.proto
buf.yaml
```

```yaml
version: v1beta1
breaking:
    ignore_unstable_packages: true
    use: [PACKAGE]
```

With `use: [PACKAGE]`, deleting one message from the package is an error, but deleting the whole package is not.

```
rm test/v1test/a.proto
buf breaking --against '.git#branch=master'
input>:1:1:Previously present message "A" was deleted from package "test.v1test".
rm test/v1test/b.proto
buf breaking --against '.git#branch=master'
// no error once the whole package is deleted
```

With `use: [FILE]`, deleting either one or both messages does not raise an error.

**Comment by bufdev** [comment]
Just a note: this was happening before #349, I just tested with a version of `buf` before that. However, this does appear to be an overall issue. Will look into it

**Comment by bufdev** [comment]
This is fixed in #414 and will go out in the next release.

**Comment by bufdev** [comment]
This is fixed on 0.50.0.
